src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java

```diff
@@ -130,19 +141,17 @@ public XSDValidation verifyXSD(MultipartFile xml) {
 
     @Transactional
     public void createIca(@NotNull MultipartFile file) {
-        // syntactic checks
-        checkSyntax(file);
+        List<CheckItem> checks = verifyIca(file);
+
+        Optional<CheckItem> check = checks.stream().filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID)).findFirst();
```

Comment:
Just a hint: put each method in a differente line. In this way in a stacktrace it's easier to understand where is the problem.

example:
```suggestion
        Optional<CheckItem> check = checks.stream()
        .filter(item -> item.getValid().equals(CheckItem.Validity.NOT_VALID))
        .findFirst();
```

Reply:
ok 👍